### ISSUE TITLE
Expose Ctrl+Return as alternate behavior for extensions commands

### ISSUE THREAD

**Issue by alefragnani** [Microsoft/vscode#18064]
When you declare a `command` for an extension, it appears in `Command Palette` and you can select/use it with `Return Key`/`Mouse Click`. But there are some _core commands_ that has _special behavior_ when you combine with `Ctrl`, like `Open on the Side`.

I would like to have the ability to declare a command and have the _normal_ and _alternative_ behaviors, just like that. 

Thanks in advance

**Comment by sandy081** [comment]
You cannot achieve that with a single command. A command will always have a single behavior and no alternate behavior. You always have to define different commands for different behaviors and can bind different key bindings to them.

**Comment by alefragnani** [comment]
Hi @sandy081 , I guess I messed up with the issue description :cry: . The intention was not for `commands`, but for `QuickPick` . I just noted my mess when I was starting to answer you to better explain my needs, sorry.  This is my real scenario:

My **Project Manager** extension has two commands: `List Projects to Open` and `List Projects to Open in New Window`. Both uses a `QuickPick` to display a list of folders. The user selects one and I open VSCode with that folder. The only difference is that the second command _opens a new VSCode instance_ instead of replacing the current one. 

The original idea of this issue was to be able to select items (and detect the modifier) with `Ctrl + Return`. Doing so, I could _remove the `List Projects to Open in New Window` command from my extension. As you can see, there was nothing to do with `Ctrl` modifier for commands, my bad :cry: 

Sorry for the misunderstanding.

**Comment by sandy081** [comment]
If I understand correctly, your requirement was to know how a user selected an entry in quick pick say
- source (mouse, key board)
- alt modifiers

**Comment by alefragnani** [comment]
Yes, exactly 👍 

I have looked at VSCode source code sometime ago, and found where this was handled. If I remember correctly, only the `Ctrl` modifier being handled (not `Shift` nor `Alt`), but this may have changed since then.

Thanks in advance

**Comment by alexrussell** [comment]
Looks like the new `alternativeAcceptSelectedQuickOpenItem` might allow for this, now...?

**Comment by TylerLeonhardt** [comment]
Although I agree that this is a valid request, I wonder if instead you'd prefer to use the new `QuickPickItemButtons` as it is more obvious what the alternative actions for a particular item are.

Sometimes I accidentally trigger the `ctrl+enter` behavior in the quick pick and worry that it could be confusing to a lot of users if they don't _know_ that `ctrl` causes an alternate behavior.

**Comment by alefragnani** [comment]
In fact, I would like to have both 😬, but yes, the new `QuickPickItemButtons` API is a great alternative and I'm definitely embracing, when available.

The reason to _prefer_ the modifier approach is because users complains about the extension _not following VS Code behavior_. Not only my **Project Manager** extension would use, but also both **Bookmarks** extensions.

But I understand your point. The issue is quite old and has only a few upvotes/interactions. If you don't see this happening in the near future, nor other devs/teams requesting such API, feel free to close the issue in favor of the `QuickPickItemButtons` API. I'm ok with that 👍 .

Thank you

**Comment by jppellet** [comment]
I agree this would be useful. I would enjoy being able to tell if e.g. Alt or Ctrl is pressed when for instance an action button atop the editor area that I contribute is pressed, and act then accordingly.

**Comment by therealsivie** [comment]
Still relevant in 2026 for the Command Palette half of this.

Worth noting the sibling gap: `TreeItem.command` has no `alt` counterpart either, even though the `menus` contribution point already supports `"alt": "<commandId>"` for inline item actions. Filed separately as microsoft/vscode#336540 so the two cases can be tracked apart.
